index.js

```diff
@@ -1,3 +1,165 @@
+            type: 'boolean'
+        }
+    })
+    .command('generate', 'generates the Dockerfile specification for the service', {
+        r: {
+            alias: 'running',
+            default: false,
+            describe: 'generate the Dockerfile to start the service',
+            type: 'boolean'
+        },
+        t: {
+            alias: 'testing',
+            default: false,
+            describe: 'generate the Dockerfile to test the service',
+            type: 'boolean'
+        },
+        b: {
+            alias: 'building',
+            default: false,
+            describe: 'generate the Dockerfile to build the deployment repository',
+            type: 'boolean'
+        },
+    })
+    .help('h')
+    .alias('h', 'help')
+        .argv;
+
+    args.deployRepo = args.deployRepo || args.review;
+    args.build = args._.indexOf('build') !== -1 || args.deployRepo;
+    args.dockerStart = args._.indexOf('docker-start') !== -1;
+    args.dockerTest = args._.indexOf('docker-test') !== -1;
+    args.generate = args._.indexOf('generate') !== -1;
+    args.deployRepo = args.deployRepo || args.build || args.building;
+
+    if ([args.build, args.dockerStart, args.dockerTest, args.generate]
+        .filter(x => !!x).length > 1) {
+        console.error('Only one command can be specified!');
+        process.exit(1);
+    }
+
+    return {
+        num_workers: args.numWorkers,
+        configFile: args.config,
+        displayVersion: args.v,
+        build: args.build,
+        buildDeploy: args.deployRepo,
+        reshrinkwrap: args.reshrinkwrap,
+        sendReview: args.review,
+        dockerStart: args.dockerStart || args.running,
+        dockerTest: args.dockerTest || args.testing,
+        generate: args.generate,
+        useDocker: args.deployRepo || args.dockerStart || args.dockerTest || args.generate,
+        force: args.force,
+        verbose: args.verbose
+    };
+}
+
+function getAppBasePath(config) {
+    if (process.env.APP_BASE_PATH) {
+        return process.env.APP_BASE_PATH;
+    } else if (config.app_base_path) {
+        return config.app_base_path;
+    } else if (/\/node_modules\/service-builder\/lib$/.test(__dirname)) {
+        // Default to guessing the base path
+        return path.resolve(`${__dirname}/../../../`);
+    } else {
+        return path.resolve('./');
+    }
+}
+
+function replaceEnvVars(config) {
+    const envRegex = /\{\s*env\(([^,\s)]+),?\s*([^)]+)?\)\s*}/g;
+    if (Buffer.isBuffer(config)) {
+        config = config.toString();
+    }
+    return config.replace(envRegex, (match, envName, defValue) => {
+        if (process.env[envName] !== undefined) {
+            return process.env[envName];
+        }
+        if (defValue !== undefined) {
+            return defValue;
+        }
+        return '';
+    });
+}
+
+function main() {
+    const options = getOptions();
+    let configFile = options.configFile;
+    if (!/^\//.test(configFile)) {
```

Comment:
@d00rman Yup, this is just the first iteration - here I wanted to just copy-paste as much as possible and then start refactoring in the next PR